File: Rohit Singhal/Arrays and Strings/ContainerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        int max=0;
        int i=0;
        int j=height.length-1;
        while(i<j)
        {
            int temp=(j-i)*Math.min(height[i],height[j]);
            if(temp>max)
            {
                max=temp;
            }
            if(height[i]<=height[j])
            {
                i++;
            }
            else
            {
                j--;
            }
        }
        return max;
        
    }
}

File: Hashwanth/Arrays & Strings/Container With Most Water/Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int a=0;
        int i=0;
        int j=height.size()-1;
        while(i<j)
        {
            int mh=min(height[i],height[j]);
            a=max(a,mh*(j-i));
            while(height[i]<=mh && i<j) i++;
            while(height[j]<=mh && i<j) j--;
        }
        return a;
    }
};
